src/main/java/org/wildfly/security/auth/login/RealmInfo.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2013 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.login;
+
+import org.wildfly.security.auth.spi.SecurityRealm;
+
+/**
+ * <p>Holds the reference to a {@link org.wildfly.security.auth.spi.SecurityRealm} and the configuration associated with it.</p>
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class RealmInfo {
```

Comment:
We probably want to have a per-realm ``NameRewriter`` here.  And I guess the role mapper will be here too when it's ready.